spring-boot/src/main/java/org/springframework/boot/logging/logback/SpringProfileAction.java

```diff
@@ -65,11 +69,12 @@ public void begin(InterpretationContext ic, String name, Attributes attributes)
 	}
 
 	private boolean acceptsProfiles(InterpretationContext ic, Attributes attributes) {
-		String profileName = attributes.getValue(NAME_ATTRIBUTE);
-		if (!OptionHelper.isEmpty(profileName)) {
-			OptionHelper.substVars(profileName, ic, this.context);
-			return this.environment != null
-					&& this.environment.acceptsProfiles(profileName);
+		String[] profileNames = StringUtils.commaDelimitedListToStringArray(attributes.getValue(NAME_ATTRIBUTE));
+		if (!CollectionUtils.isEmpty(Arrays.asList(profileNames))) {
```

Comment:
It'd be better to check for `profileNames.length == 0` rather than creating and immediately throwing away the list